### ISSUE TITLE
SelectList not executing the callback set for SelectedValueChanged

### ISSUE THREAD

**Issue by baltazarO** [Megabit/Blazorise#5594]
### Blazorise Version

1.5.3

### What Blazorise provider are you running on?

None

### Link to minimal reproduction or a simple code snippet

`<SelectList 
                    DefaultItemDisabled 
                    DefaultItemHidden=true 
                    DefaultItemText="Pick a data type" 
                    TItem="DataType" 
                    Data="Foo.DataTypes" 
                    TValue="int" 
                    ValueField="@((x) => x.ID)" 
                    TextField="@((x) => x.NameOfType)" 
                    SelectedValue="Foo.Bar.DataTypeID"
                    SelectedValueChanged="(e) => TheDataTypeChanged(e)"/>`

`    void TheDataTypeChanged(int newValue)
    {
        Foo.Bar.DataTypeID = newValue;
        Foo.CallStateHasChanged();
    }`

### Steps to reproduce

1. Set a breakpoint in the "TheDataTypeChanged" method
2. First you need to create a class called Foo that has a list of datatypes and a Bar object that is = new Bar();
3. Make sure datatypes is initialized (DataType class has an ID and a Name)
4. Choose a datatype from the select list dropdown

### What is expected?

I should hit the breakpoint in the "TheDataTypeChanged" method

### What is actually happening?

Never hits the breakpoint.

### What browsers do you see the problem on?

Chrome

**Comment by David-Moreira** [comment]
I don't see a visible issue with your code.
There seems to be no issue with binding by going to the docs : https://blazorise.com/docs/extensions/selectlist

May I ask that you provide a more complete piece of code? At least the whole component or page? With everything that's required to run your reproducible successfully? Thanks!

**Comment by lfoust** [comment]
I am hitting this same issue. I have uploaded a complete example based on the repro posted above. Project attached.

[SelectListRepro.zip](https://github.com/user-attachments/files/16325089/SelectListRepro.zip)

**Comment by stsrki** [comment]
> Certainly Blazorise can possibly improve the experience with automatically binding complex types, but unfortunately as far as I remember this is still a known limitation because blazorise uses native html select and I believe there were limitations with being able to properly map natively as it's just not supported by the browser, correct me if I'm wrong [@stsrki](https://github.com/stsrki)

Yes, that is correct. The `TValue` must map to a known primitive type in order for it to be able to render to a native HTML value.

Thanks @David-Moreira

**Comment by zl-aric** [comment]
Why not add a TKey attribute for the complex type?
